### ISSUE TITLE
Double quotes no longer working as intended

### ISSUE THREAD

**Issue by wickles** [DestinyItemManager/DIM#3431]
It seems to be picking up hits as if a wildcard is implied between words. 

Example: I search "grenade launcher reserves", and it finds hits with Grenade Launcher Scavenger (**Grenade Launcher**s get bonus **reserves**...) but no Grenade Launcher Reserves. 

I suspect this may be related to https://github.com/DestinyItemManager/DIM/pull/3386.

**Comment by bhollis** [comment]
@48klocs

**Comment by delphiactual** [comment]
@wickles is the just `text search` or off of `perk:`?

**Comment by 48klocs** [comment]
We could tighten it up to simply search for the perk name (or words in a row), but that could cause regressions for other folks, so I'm hesitant to further restrict it.

**Comment by philkernick** [comment]
That implies that perk:"grenade launcher reserves" is really just perk:grenade perk:launcher perk:reserves

That is definitely not expected.  I'm with the OP.  This is a bug that should be fixed.

**Comment by DHager** [comment]
Encountered same issue in the wild, it makes it difficult to find specific perks e.g. `perk:"sniper rifle reserves"`

**Comment by philkernick** [comment]
What we really need is that perk searches anchors at a word boundary. Whether it’s a single word or a double quoted string.

**Comment by DHager** [comment]
Anchoring at word boundaries? I don't understand, what problem is solved by adding that unexpected behavior? 

I might not remember where the park is called fletching or fletched, I just want the bow with the substring `fletch`

**Comment by delphiactual** [comment]
@DHager https://github.com/DestinyItemManager/DIM/pull/3386#issuecomment-448977910

**Comment by philkernick** [comment]
@DHager what you've said is exactly what anchoring at a word boundary means.

What we are trying to avoid is perk:hip matching something with "ship" in it.

**Comment by DHager** [comment]
I see one of the scenarios you're trying to solve -- matching "machine gun reserves" without matching "*sub*machine gun reserves", and I agree that needs a solution.

However using "match whole words only" behavior as the default is undesirable, because it's not what users expect. They will expect a straight substring match for `foo`, because it's what all the other software they use every day has taught them to expect:
 
* Searching for text in browser page
* Searching in word-processor / spreadsheet tools
* Searching in text-editors
* Searching in Google (which does lots of stop-words and stemming and fuzzy matching)

In all of those cases, "match whole words only" is not the default, and must be opted-in.

**Comment by 48klocs** [comment]
@bhollis in spite of the bickering over what color to paint this particular bike shed, I think the way it works is fine.

My intent was to be able to search for `machinegun` or `io` without having to perform increasingly arcane incantations, and it does that.

I suspect not that many folks think to wrap search terms in quotes (I sure don't!); those who know enough to do the quote thing probably also know enough to add negation terms to filter things out as they want, so enjoy going hog wild, friends.

**Comment by philkernick** [comment]
Amen to that.  perk:io should not match "faction".

**Comment by philkernick** [comment]
Bike shed painting equipment here.

It's never been "match whole words".  It's anchor match at the start of words.  That is completely different.

And if you really want free search, don't use the perk: filter.  Ordinary search has always worked this way.

**Comment by DHager** [comment]
Given that this bug is negatively impacting users, how about at least reverting it to whatever the prior behavior was?

**Comment by philkernick** [comment]
I agree with reverting to previous behaviour, but with a minor move forward as well.

Assuming Perl compatible regular expressions, then the perk search should have an implied \b at the beginning to anchor it on a word boundary.

**Comment by bhollis** [comment]
Agree with @philkernick. Posted up a fix at https://github.com/DestinyItemManager/DIM/pull/3484.
